src/core/dynamic-user.c

```diff
@@ -641,18 +655,25 @@ int dynamic_user_serialize(Manager *m, FILE *f, FDSet *fds) {
         /* Dump the dynamic user database into the manager serialization, to deal with daemon reloads. */
 
         HASHMAP_FOREACH(d, m->dynamic_users)
-                (void) dynamic_user_serialize_one(d, "dynamic-user", f, fds);
+                (void) dynamic_user_serialize_one(d, "dynamic-user", f, fds, /* index= */ NULL);
 
         return 0;
 }
 
-void dynamic_user_deserialize_one(Manager *m, const char *value, FDSet *fds, DynamicUser **ret) {
+void dynamic_user_deserialize_one(
+                Manager *m,
+                const char *value,
+                FDSet *fds,
+                int *fds_array,
+                size_t n_fds_array,
+                bool store_index,
```

Comment:
if the fdset is recognizable as indexed or not via fdset_is_indexed() (as suggested before) you could drop this part